larray/inout/pandas.py

```diff
@@ -239,8 +248,11 @@ def df_aslarray(df, sort_rows=False, sort_columns=False, raw=False, parse_header
                 # (see https://github.com/pandas-dev/pandas/issues/15105)
                 df = series.unstack(level=-1, fill_value=fill_value)
                 # pandas (un)stack and pivot(_table) methods return a Dataframe/Series with sorted index and columns
-                labels = df_labels(series, sort=False)
-                index = pd.MultiIndex.from_tuples(list(product(*labels[:-1])), names=series.index.names[:-1])
+                labels = index_to_labels(series.index, sort=False)
+                if isinstance(df.index, pd.core.index.MultiIndex):
```

Comment:
This code is getting really ugly. Couldn't some of this code be transferred to from_series to support multiIndex and here only use something like:
```python
series = df[df.columns[-1]]
return from_series(sort_rows=sort_columns)
```